### ISSUE TITLE
E208 should not trigger on symlinks

### ISSUE THREAD

**Issue by robinmulder** [ansible/ansible-lint#981]
##### Summary

Currently E208 also triggers on symlinks (`state: link` with the file module). A symlink always has mode 777. If you set a mode in Ansible on a symlink it will set those permissions on the source file, and not on the symlink itself. Also, `mode: preserve` is not possible with a symlink.

In my opinion this rule should be skipped when `state: link` is set since this doesn't offer any benefits and only adds confusion. This also doesn't solve any (potential) security issue which was the original reason (as far as I can see) for implementing this rule. In fact, it introduces potential security issues since someone might set mode 777 on the symlink (since that is the default mode for a symlink) due to this rule, not realizing that those permissions will be set on the source file instead of the symlink.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
ansible 2.8.14
ansible-lint 4.3.1
```

##### STEPS TO REPRODUCE

```
---
- hosts: localhost

  tasks:
    - name: create file
      file:
        path: test
        state: touch
        mode: 0644

    - name: symlink file
      file:
        path: test2
        src: test
        state: link
```
